### DISCUSSION TITLE
flash set GPON_SN with \0

### DISCUSSION THREAD

**Discussion by shdf2** [Anime4000/RTL960x#80] [Q&A] [answered]
Hello,
my GPON_SN is 12 char and i have to send 16 char to the OLT, i must fill 4 ascii 0 char at the end of the string, i saw [here](https://forum.mikrotik.com/viewtopic.php?t=116364&start=200#p751330) that they add **\0** at the end of the string so i tried 
flash set GPON_SN XXXXXXXXXXXX\0\0\0\0
i checked with a flash get GPON_SN and it returns XXXXXXXXXXXX without the \0\0\0\0 maybe it's normal as the chars can't be displayed, but even if i receive an O5 i can't connect to internet so i guess this didn't work.

Anyone can help ?

**Comment by Anime4000** [comment, marked as answer]
RTL960x is using: 4 ASCII + 8 HEX
Example: HWTC1234ABFF

if your original ONU has 16 HEX, you need to convert first 8 HEX to 4 ASCII, then combine converted 4 ASCII + 8 HEX

it always 4 ASCII + 8 HEX, no need to add `\0` that for Lantiq

**Comment by shdf2** [comment]
Thanks, so my problem is still elsewhere...
